ERCS/erc-7753.md

```diff
@@ -0,0 +1,88 @@
+---
+eip: 7753
+title: Algorithmically generated NFT
+description: The image of the NFT is represented through mathematical algorithms.
+author: LEVI_104 (@chen4903), birch (@birchwork)
+discussions-to: https://ethereum-magicians.org/t/algorithmically-generated-nft/20761
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-08-08
+requires: 165, 721
+---
+
+## Abstract
+
+A non-fungible token standard that uses mathematical algorithms to draw and store images without any servers or storage space.
+
+## Motivation
+
+The current NFT solutions store images on IPFS or third-party servers, consuming storage space and being separate from the blockchain. Our goal is to represent image information using mathematical algorithms stored in smart contracts, tightly integrating with the blockchain. This approach uses no storage space and allows anyone to render the image information at any time based on the algorithm stored in the smart contract.
+
+## Specification
+
+### Interface
+
+This standard requires the implementation of the following interface:
+```solidity
+pragma solidity ^0.8.0;
+
+/// @title ERC-7790 Algorithmically Generated Token Standard
+interface IERC7790 {
+    /// @dev When a new token is created, emit this event to record the algorithm content.
+    event SetTokenAlgorithm(uint256 indexed tokenID, string algorithm);
+
+    /// @dev The project team or administrator manually inputs the mathematical 
+    /// algorithm for the corresponding token, with the parameter algorithm adhering 
+    /// to the format requirements specified by this EIP.
+    function setTokenAlgorithm(uint256 tokenID, string memory algorithm) external returns(bool);  
+    
+    /// @dev Returns the mathematical algorithm for the token image based on
+    /// the token ID. The returned data is a JSON text that adheres to the format specified by this EIP.
+    function getTokenAlgorithm(uint256 tokenID) external view returns(string memory);
+}
+```
+### Format Requirements
+
+```json
+{
+    "Name": "Asset name",
+    "Encoding Method": "Specify the string encoding and decoding method for storing the algorithm",
+    "properties": [
+       "Mathematical Formula 1",
+       "Mathematical Formula 2",
+       "...", 
+    ]
+}
+```
```

Comment:
I'd recommend using [JSON Schema](https://json-schema.org/) to formally specify what the JSON is supposed to look like.